docs/features/network-security-controls/admin-network-policy.md

```diff
@@ -1,303 +1,1118 @@
+    * rule4: deny all other ingress traffic
+* `egress` has 6 rules:
+    * rule0: always allow egress traffic coming from all pods to dns pods
+    * rule1: always allow TCP egress traffic coming from all pods to KAPI pods
+    * rule2: always allow all egress traffic coming from all pods to splunk logging
+      tenant at TCP ports 8991 and 8992
+    * rule3: always allow all egress traffic to `open` tenant's namespaces, company's
+      intranet CIDR ranges and all worker nodes in the cluster
+    * rule4: always pass all egress traffic to `restricted` tenant's namespaces
+    * rule5: deny all other egress traffic
+
+You can do `oc apply -f cluster-control.yaml` to create this ANP in your cluster
+```shell
+$ oc get anp
+NAME              PRIORITY   AGE
+cluster-control   34         3s
+```
 
-* Each AdminNetworkPolicy CRD will have a `.spec.priority` field. The lower the number the higher the precedence. Thus 0 is the highest priority and 1000 (the largest number supported by upstream sig-network-policy-api) is the lowest priority. However the number of admin policies in a cluster are usually expected to be of a maximum of say 30-50 and not more than that based on use cases for which this API was defined for. OVNK plugin will support 100 policies max in a cluster. If anyone creates more than 100, it will not work properly. Thus supported priority values in OVNK are from 0 to 99.
-* Each AdminNetworkPolicy CRD can have upto 100 ingress rules and 100 egress rules, thus 200 rules in total. The ordering of each rule is important. If the rule is at the top of the list then it has the highest precedence and if the rule is at the bottom it has the lowest prededence. Each rule translates to one ACL.
-* Since we can have upto 100 policies and each one can have upto 100 gress rules (100*100), we have blocked out the range: 30,000 - 20,000 priority range for the OVN nbdb.ACL table in the `Tier1` block for ANP's implementation.
-* Each AdminNetworkPolicy CRD will have a subject on which the policy is applied on - this is translated to one PortGroup on which the ACLs of each rules are attached on.
-* Each gress rule can have upto 100 peers. Each rule will also create an nbdb.AddressSet which will contain the IPs of all the pods that are selected by the peer selector across all the peers of that given rule.
+#### BaselineAdminNetworkPolicy Sample API
 
-The PortGroup for the above AdminNetworkPolicy is:
+Unlike ANP's BANP is a singleton object in your
+cluster meant to create a guardrail default security
+posture.
 
+Let us look at a sample BANP yaml:
+
+```yaml
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: BaselineAdminNetworkPolicy
+metadata:
+  name: default
+spec:
+  subject:
+    namespaces: {}
+  ingress:
+  - name: "default-deny"
+    action: "Deny"
+    from:
+    - namespaces:
+        namespaceSelector: {}
+  egress:
+  - name: "default-deny"
+    action: "Deny"
+    to:
+    - namespaces:
+        namespaceSelector: {}
 ```
```

Comment:
You should probably explain this: that it blocks all traffic between all pods in the entire cluster by default, and requires everybody to create _both_ ingress and egress policies for all of the traffic that they want to allow.

Also, it applies even in private openshift namespaces that the cluster-admins don't know anything about, so probably you don't really want this policy...

Reply:
yeah I have added a point in best practices bit to never use `{}` selectors but will clarify it here as well.